src/fastfield/serializer/mod.rs

```diff
@@ -33,8 +36,45 @@ use crate::schema::Field;
 /// * `close()`
 pub struct CompositeFastFieldSerializer {
     composite_write: CompositeWrite<WritePtr>,
+    codec_enable_checker: FastFieldCodecEnableCheck,
+}
+
+pub struct FastFieldCodecEnableCheck {
+    enabled_codecs: Vec<FastFieldCodecName>,
+}
+impl FastFieldCodecEnableCheck {
+    fn allow_all() -> Self {
+        FastFieldCodecEnableCheck {
+            enabled_codecs: ALL_CODECS.to_vec(),
+        }
+    }
+    fn is_enabled(&self, codec_name: FastFieldCodecName) -> bool {
+        self.enabled_codecs.contains(&codec_name)
+    }
 }
 
+impl From<FastFieldCodecName> for FastFieldCodecEnableCheck {
+    fn from(codec_name: FastFieldCodecName) -> Self {
+        FastFieldCodecEnableCheck {
+            enabled_codecs: vec![codec_name],
+        }
+    }
+}
+
+pub const FF_HEADER_MAGIC_NUMBER: u8 = 123u8;
```

Comment:
why 123?

Reply:
I think that 123 is less probable on data/memory corruption